### DISCUSSION TITLE
Implementing Eizo monitor control

### DISCUSSION THREAD

**Discussion by markbolhuis** [rockowitz/ddcutil#460] [Ideas]
At the moment, ddcutil does not handle Eizo monitors, which is a problem for programs that use ddcutil, like KDE Powerdevil.

Eizo use the HID protocol to communicate with their monitor. But does so with vendor usage pages, and also does so in a way that isn't HID compliant.

I've taken it upon myself to reverse engineer the usages and protocol, and I'm writing a C library [libeizo](https://github.com/markbolhuis/libeizo) to control these monitors. This library is still a WIP but I've discovered the function of a substantial number of these vendor defined usages, enough to cover 95% of use cases. You can see these usages [here](https://github.com/markbolhuis/libeizo/blob/6ad814649a3545934ae83300a96a854aa455ad72/src/internal.h#L18).

I'm opening this discussion to talk about how ddcutil can be expanded to include Eizo monitors by default. 

Thank You
Mark

**Comment by rockowitz** [comment]
Impressive!  How did you reverse engineer the usages?  Sniff the USB bus?

I have to admit I've considered eliminating the USB support.  Given that there's never a bug report, I have the impression that it's hardly being used. That said, what you've done would make the USB implementation much more useful, particularly if it could be generalized to reverse engineer the control for any display that uses USB to communicate with the monitor's VCP.  Even better would be the ability to load using usage tables at runtime, perhaps as an extension of user defined feature files. 

It's been several years since I implemented the USB support, and I learned just enough about USB to get the job done, but I think it would be straightforward to merge in the Eizo support.  I'd welcome it into ddcutil.

One qualification.  While it may be the case that ddcutil does not support your monitor, it's not entirely true that ddcutil does not support Eizo monitors.  The ddcutil USB code was developed using two different Eizo monitors that followed, at least in part, the USB HID Monitor spec.  One, IIRC, was a CG19, that used solely used USB.   The other, whose model I don't immediately recall, could communicate using either USB or DDC.

Regards, 
Sanford
